build-tools/docker/example-mainnet/docker-compose.yml

```diff
@@ -0,0 +1,132 @@
+version: '3'
+
+x-common: &ml-common
+  volumes:
+    - ./mintlayer-data:/home/mintlayer
+
+x-common-env: &ml-common-env
+  RUST_LOG:
+  ML_USER_ID:
+  ML_GROUP_ID:
+
+services:
+  node-daemon:
+    <<: *ml-common
+    image: $DOCKERHUB_USERNAME/node-daemon:$ML_SOFTWARE_VERSION
+    command: node-daemon mainnet
+    environment:
+      <<: *ml-common-env
+      # Note:
+      # The default rpc bind address is '127.0.0.1', so it'll only be reachable from the same
+      # container; to make it reachable from other containers, we need to set it to '0.0.0.0'.
+      # For p2p, the default address is already '0.0.0.0'. But it's better to use the same port
+      # as the one exposed on the host system, otherwise other nodes won't be able to establish
+      # outbound connections to this node.
+      # Details: the node determines its own "public" p2p address by combining its ip address
+      # as it is seen by a peer with the port that it is listening on; this address is then
+      # advertised to other peers and propagated through the network. So, if the listening port
+      # differs from the one exposed on the host, the advertised address will be incorrect.
+      ML_MAINNET_NODE_RPC_BIND_ADDRESS: 0.0.0.0:3030
+      ML_MAINNET_NODE_P2P_BIND_ADDRESSES: 0.0.0.0:$NODE_P2P_HOST_PORT
+      # Rpc username and password.
+      ML_MAINNET_NODE_RPC_USERNAME: $NODE_RPC_USERNAME
+      ML_MAINNET_NODE_RPC_PASSWORD: $NODE_RPC_PASSWORD
+    ports:
+      # This is only needed if you want to access the node's rpc interface from the host system.
+      - "$NODE_RPC_HOST_PORT:3030"
+      # This is only needed if you want the node to be able to accept incoming p2p connections
+      # from other nodes.
+      - "$NODE_P2P_HOST_PORT:$NODE_P2P_HOST_PORT"
```

Comment:
This is some weird port binding... I would understand if 3031 is fixed then you choose the exposed port...

Reply:
[discussed] there is a comment above that explains why this is needed. I've moved it closer to this line to make things clearer.